pages/guides/developing/encrypted-vod.en-US.mdx

```diff
@@ -0,0 +1,160 @@
+
+First, we generate a 256-bit encryption key to encrypt the file. 
+
+```js
+// Generate a random 256-bit key
+const key = await window.crypto.subtle.generateKey(
+	{
+		name: "AES-CBC",
+		length: 256,
+	},
+	true,
+	["encrypt", "decrypt"]
+);
+
+// Export the key as raw data
+const keyData = await window.crypto.subtle.exportKey("raw", key);
+
+// Encode the key in Base64
+const keyBase64 = btoa(String.fromCharCode(...new Uint8Array(keyData)));
+```
+
+## Step 2: Encrypt your video using the encryption key
+
+To encrypt a video file with the key we just generated, we follow these steps:
+
+1. Generate a random 128-bit initialization vector (IV).
+2. Pad the data using PKCS#7 padding.
+3. Encrypt the data using AES-CBC.
+
+```js
+const iv = window.crypto.getRandomValues(new Uint8Array(16));
+
+const encrypted = await window.crypto.subtle.encrypt(
+	{
+		name: "AES-CBC",
+		iv: iv,
+	},
+	key, // from generateKey or importKey above
+	arrayBuffer // ArrayBuffer of data you want to encrypt
+);
+
+// Concatenate IV and encrypted file into a new ArrayBuffer
+const resultBuffer = new ArrayBuffer(iv.length + encrypted.byteLength);
+new Uint8Array(resultBuffer).set(new Uint8Array(iv), 0);
+new Uint8Array(resultBuffer).set(new Uint8Array(encrypted), iv.length);
+
+const blob = new Blob([resultBuffer], { type: "application/octet-stream" });
+```
+<Callout>
+	Currently Livepeer supports only video content encrypted using [SubtleCrypto.encrypt](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/encrypt) with `AES-CBC` algorithm, which is also the encryption used by other web3 protocols like [Lit](https://developer.litprotocol.com/sdk/explanation/encryption/). It can implemented in other environments with regular AES-CBC encryption using PKCS#7 padding.
+</Callout>
+
+## Step 3: Retrieve the Livepeer Public Key
+
+After obtaining an encryption key and encrypting a video file with it, the next step is to retrieve the Livepeer Public Key. This key will be used to encrypt our encryption key, which can then be shared with Livepeer when creating a video asset.
+
+```js
+// Fetch the public key from Livepeer
+const publicKeyResponse = await fetch('https://livepeer.monster/api/access-control/public-key', {
+	headers: {
+		'Authorization': 'Bearer XXXX-XXXXXX-XXXXXXX-XXXX',
+	},
+});
+
+const publicKeyData = await publicKeyResponse.json();
+const publicKey = publicKeyData.public_key;
+```
+
+## Step 4: Encrypt the encryption key
+
+Now, we can use asymmetric encryption with the Livepeer Public Key to encrypt our Encryption Key. The resulting data is then encoded in base64 before being sent to Livepeer along with the video file.
+
+<Callout>
+	The Web Cryptography API built into modern browsers does not directly support PEM or PKCS#1 formatted keys. It only supports the SPKI format for public keys and the PKCS8 format for private keys. This is an example of a server-side Golang handler receiving the encryptionKey and the publicKey.
+</Callout>
```

Comment:
Let's remove this callout and write the JS example below assuming the public key comes in the format supported by JS. Then we can get this guide out of the way and make the fix to the frontend async/as a followup, wdyt?